### PR TITLE
Update Makefile.in to allow cross-compilation

### PR DESCRIPTION
**Pull request by ffontaine**
Use `DESTDIR` and `LDFLAGS` environment variables passed to `configure` in `Makefile.in` files

Signed-off-by: Fabrice Fontaine <fabrice.fontaine@orange.com>
[Retrieved (and slightly updated) from:
https://git.buildroot.net/buildroot/tree/package/tinydtls/0001-Update-Makefile.in-to-allow-cross-compilation.patch]

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>

### CODE REVIEW

File: Makefile.in [hunk 1]
```diff
@@ -52,8 +52,8 @@ DISTDIR=$(top_builddir)/$(package)
 FILES:=Makefile.in configure configure.ac dtls_config.h.in \
   Makefile.tinydtls $(SOURCES) $(HEADERS)
 LIB:=libtinydtls
-LIBS:=$(LIB).a $(LIB).so
-LDFLAGS:=@LIBS@
+LIBS:=$(LIB).a $(LIB).so @LIBS@
```
Comment (obgm) [inline Makefile.in:55]: I think that `@LIBS@` must be in `LDFLAGS`. It is set by `configure` on platforms that have `-liblnsl -libsocket` (e.g., SunOS). The Makefile variable `LIBS`, on the other hand, contains the make targets governed by this Makefile.

File: Makefile.in [hunk 2]
```diff
@@ -111,10 +111,10 @@ dist:	$(FILES) $(DISTSUBDIRS)
 	tar czf $(package).tar.gz $(DISTDIR)
 
 install:	$(LIBS) $(HEADERS) $(SUBDIRS)
-	test -d $(libdir) || mkdir -p $(libdir)
-	test -d $(includedir) || mkdir -p $(includedir)
-	$(install) $(LIBS) $(libdir)/
-	$(install) $(HEADERS) $(includedir)/
+	test -d $(DESTDIR)$(libdir) || mkdir -p $(DESTDIR)$(libdir)
```
Comment (obgm) [inline Makefile.in:114]: I do not understand why `DESTDIR` would be used here and at the other places specified. The unfolding of `libdir` and `includedir` follows the common GNU autoconf pattern to:

```
exec_prefix = ${prefix}
libdir = ${exec_prefix}/lib
includedir = ${prefix}/include/tinydtls
```

Given that `prefix` can be set to contain whatever you might want to set `DESTDIR` to, I do not see why you need another prefix?

File: Makefile.in (hunk 1 above)
Comment (ffontaine) [reply Makefile.in:55]: OK, I'll update the PR accordingly

File: Makefile.in (hunk 2 above)
Comment (ffontaine) [reply Makefile.in:114]: More information on `DESTDIR` can be found here: 
- https://www.gnu.org/prep/standards/html_node/DESTDIR.html
- https://www.gnu.org/software/automake/manual/html_node/DESTDIR.html

Basically, `DESTDIR` is used to set the path where the different component will be installed on the _local_ machine (e.g. `/home/fabrice/buildroot/output/target/`) whereas prefix and exec_prefix must be set to the path on the _target_ machine (e.g. `/usr`). As explained in the second link above: 

> DESTDIR support is strongly recommended in GNU packages, though it is not an absolute requirement.

Comment (obgm) [reply Makefile.in:114]: Sorry, I have missed this.

**Comment by obgm** [comment]
Thank you. For this PR to be merged you will need to create and electronically sign the Eclipse Contributor Agreement (ECA) and sign off on the Eclipse Foundation Certificate of Origin.
I assume that every author email address mentioned in the Signed-off must have the ECA signed with the particular email address used.

For more information, please visit
http://wiki.eclipse.org/Development_Resources/Contributing_via_Git

**Comment by obgm** [comment]
Please be aware that by your contribution to the Eclipse tinydtls project, you are changing the license of the orignal patch which appears to have been under GPL to [EPL](https://www.eclipse.org/legal/epl-v10.html) and [EDL](https://www.eclipse.org/org/documents/edl-v10.php).

**Comment by ffontaine** [comment]
Both ECA are signed but I'm still getting the following error:

> Fabrice Fontaine (fontaine.fabrice@****.com) did not include the "Signed-off-by footer" which is required for all commits made by a contributor.

which is obviously wrong but perhaps the system is not able to manage a PR with two Signed-off-by. I'll remove the first one to check if tit fixes the issue.

**Comment by ffontaine** [comment]
And indeed, with only one Signed-off-by, it passes. Tell me if this ok for you.

**Comment by obgm** [comment]
Looks good. Not sure why there was an issue. It should have worked as long as the ECA has been signed for the email addresses of every Signed-off-Footer. At least it did before...

**Comment by ffontaine** [comment]
Then, perhaps the issue was linked to the order of the Signed-off-by? The author of the PR must perhaps be the first one? If this is important, I can retry to update the PR.

**Comment by obgm** [comment]
Or the problem came from the fact that there was text in between the list of Signed-offs. I can try adding the second email address when merging, if you like, or leave it as is. Your choice.

**Comment by ffontaine** [comment]
If you can add the second address when merging, it would be perfect, thanks.

**Comment by obgm** [comment]
Merged with additional email address as commit 3b03c38.

**Pull request closed without merging**